src/frontend/hooks/useSyncPlayStreakWithExternalSource.ts

```diff
@@ -0,0 +1,104 @@
+import { useAccount, useConnect, useSignMessage } from 'wagmi'
+import useAuthSession from './useAuthSession'
+import { injected } from 'wagmi/connectors'
+import alertStore from 'frontend/store/AlertStore'
+import { useTranslation } from 'react-i18next'
+import extensionState from 'frontend/state/ExtensionState'
+import { useSyncPlayStreakWithExternalSource as useSyncPlayStreakWithExternalSourceCore } from '@hyperplay/quests-ui'
+import authState from 'frontend/state/authState'
+
+// @dev: this hook needs to be used in a component with MobX observer
+export function useSyncPlayStreakWithExternalSource({
+  refreshPlayStreak
+}: {
+  refreshPlayStreak: () => void
+}) {
+  const { isSignedIn } = useAuthSession()
+  const { address } = useAccount()
+  const { signMessageAsync } = useSignMessage()
+  const { connectAsync } = useConnect()
+  const { t } = useTranslation()
+
+  const showWalletWarning = () => {
+    alertStore.setAlert(
+      'warning',
+      t(
+        'quests.playstreak.walletNotConnected',
+        'Please connect a wallet to sync your progress.'
+      )
+    )
+  }
+
+  const { mutateAsync } = useSyncPlayStreakWithExternalSourceCore({
+    getCSRFToken: async () => {
+      return window.api.getCSRFToken()
+    },
+    syncPlayStreakWithExternalSource: async ({ quest_id, signature }) => {
+      return window.api.syncPlayStreakWithExternalSource({
+        quest_id,
+        signature
+      })
+    },
+    checkPendingSync: async ({ wallet, questId }) => {
+      return window.api.checkPendingSync({ wallet, questId })
+    },
+    signMessage: async (message) => {
+      return signMessageAsync({ message })
+    },
+    mutationOptions: {
+      onSuccess: async () => {
+        refreshPlayStreak()
+        console.log('Playstreak synced with external source')
+        window.api.logInfo('Playstreak synced with external source')
+      },
+      onError: (error) => {
+        console.error(`Error syncing playstreak with external source`, error)
+        window.api.logError(
+          `Error syncing playstreak with external source: ${error.message}`
+        )
+      }
+    }
+  })
+
+  const syncPlayStreakWithExternalSource = async (questId: number) => {
+    if (!isSignedIn) {
+      authState.openSignInModal()
+      return
+    }
+
+    const currentProvider = await window.api.getConnectedProvider()
+    const web3Provider = await window.api.getCurrentWeb3Provider()
+
+    const isWalletUnConnected =
+      currentProvider === 'Unconnected' && !web3Provider
+
+    // we do window.api.focusMainWindow() instead of onboardingStore.openOnboarding()
+    // because this can be called from a game window, not the main window
+    if (isWalletUnConnected) {
+      showWalletWarning()
+      window.api.focusMainWindow()
+      window.api.openOnboarding()
+      return
+    }
```

Comment:
I don't think this if statement will ever execute. I tried disconnecting the wallet and signing out and in. The external sync button did not show unless I was signed in and had my wallet connected

Reply:
you're right! we don't need this anymore